### PR TITLE
Remove dependency on postcss

### PR DESCRIPTION
**Pull request by jbraithwaite**
`postcss` is a pretty heavy weight dependency if it is only being used for parsing an AST.

Replaced with [css-tree](https://www.npmjs.com/package/css-tree).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -299,15 +299,31 @@ function sanitizeHtml(html, options, _recursing) {
...
+                    var rules = selectors[node.property];
+
+                    if (rules !== undefined && rules.every(function (rule) { return !value.match(rule); })) {
+                      list.remove(item)
+                    }
+                  }
+                })
+
+                value = csstree.generate(ast).slice(name.length + 1);
+                value = value.slice(0, value.length - 1);
```
Comment (squirly) [inline src/index.js]: You may want to extract 302-317 to a function.

**Comment by jbraithwaite** [comment]
From their readme:

> NOTE: The project is in alpha stage since some parts need further improvements, AST format and API are subjects to change. However it's stable enough and used by packages like CSSO (CSS minifier) and SVGO (SVG optimizer) in production.

npm reports 8 million downloads a month for this package and it is depended on [by these packages](https://www.npmjs.com/browse/depended/css-tree) (including SVGO and CSSO).

**Comment by jbraithwaite** [comment]
I can write additional tests if you think that's required.

**Comment by boutell** [comment]
This has been rolled back for now (1.19.3), pending further examination of #269, please take a look. Thanks!

**Pull request merged**